scylla-cdc-lib/src/main/java/com/scylladb/cdc/lib/CDCConsumer.java

```diff
@@ -97,12 +97,37 @@ public static Builder builder() {
         private final WorkerConfiguration.Builder workerConfigurationBuilder
                 = WorkerConfiguration.builder();
 
-        private RawChangeConsumerProvider consumerProvider;
         private int workersCount = getDefaultWorkersCount();
         private ScheduledExecutorService executorService;
 
+        @SuppressWarnings("deprecation")
         public Builder withConsumerProvider(RawChangeConsumerProvider consumerProvider) {
-            this.consumerProvider = Preconditions.checkNotNull(consumerProvider);
+            withConsumer(consumerProvider.getForThread(0));
```

Comment:
Are we losing the guarantee that each consumer is called always on the same thread?

Reply:
Yes. 100%. As we discussed previously, consumers are encouraged to be stateless and/or thread safe.